src/app.ts

```diff
@@ -6,15 +6,31 @@ import { register } from './register'
 import { AnyJson } from "@polkadot/types/types";
 import bodyParser from 'body-parser';
 import locks from "locks";
+import expressRateLimit from 'express-rate-limit';
+
+// per IP rate limit
+const expressIpLimiter = expressRateLimit({
+  windowMs: 48 * 60 * 60 * 1000,
+  max: 3, // Limit each IP to N requests per `window`
+  standardHeaders: false, // Return rate limit info in the `RateLimit-*` headers
+  legacyHeaders: true, // Disable the `X-RateLimit-*` headers
+})
 
 const app = express();
-const port = process.env.PORT || 3002;
+const port = parseInt(process.env.PORT || '3002');
 const joy = new JoyApi();
 app.use(bodyParser.json())
 
 const processingRequest = locks.createMutex();
 
 app.use(cors());
+
+// Configure number of hops behind reverse proxy
+app.set('trust proxy', 1)
```

Comment:
This should be fine as long as, for example, `proxy_set_header X-Forwarded-For $remote_addr;` is set in Nginx.
If `proxy_set_header X-Forwarded-For` is not set, I think the ip can be spoofed by the client.

Reply:
I'm running the faucet behind `caddy` I'll double check if that header is sent when it is forwarding the request.

Reply:
"For these X-Forwarded-* headers, by default, Caddy will ignore their values from incoming requests, to prevent spoofing." - source https://caddyserver.com/docs/caddyfile/directives/reverse_proxy#defaults